AK/Random.h

```diff
@@ -20,12 +20,21 @@
 #    include <sys/random.h>
 #endif
 
+#if defined(AK_OS_HAIKU)
+#    include <fcntl.h>
+#    include <unistd.h>
+#endif
+
 namespace AK {
 
 inline void fill_with_random([[maybe_unused]] Bytes bytes)
 {
 #if defined(AK_OS_SERENITY) || defined(AK_OS_ANDROID)
     arc4random_buf(bytes.data(), bytes.size());
+#elif defined(AK_OS_HAIKU)
+    int fd = open("/dev/random", O_RDONLY);
```

Comment:
We call this from enough places that we probably don't want to be opening and closing an fd to /dev/random all the time. Is there a non-file-based API we can use to grab random data on Haiku?

Reply:
There's no other way to get random data on Haiku.
The Haiku port of Epiphany/GNOME Web does it the same way: https://github.com/haikuports/haikuports/blob/50c1dcbc9ddacc98d9a2dade02cdc3192f348b49/www-client/epiphany/patches/epiphany-43.1.patchset#L54
I searched through the Haiku-specific API documentation and asked on their Matrix chat.
Something like arc4random_buf() or getentropy() doesn't seem to exist on Haiku.

Reply:
There was some work on such an API a while ago but it wasn't complete, and needed kernel changes. Some of those happened in the interim, I'll see about implementing it today.

Reply:
Implementation of getentropy/arc4random merged into Haiku master (hrev57247). It'll be in tonight's builds. So, this is now unneeded.

Reply:
Thank you very much for implementing these features so quickly.
I updated the code to make use of arc4random.

@ADKaster Maybe off-topic here, but I wonder if it would be better to make use of arc4random on the BSDs that support it instead of using getentropy, which is called a fallback in the code?
If yes, I could make that change in another pull request.

Reply:
> I wonder if it would be better to make use of arc4random on the BSDs that support it instead of using getentropy

That's probably a good change, but I don't have the BSD envs handy that you do to test such a change :)